numpy/core/tests/test_ufunc.py

```diff
@@ -1197,6 +1197,7 @@ def test_reduce_arguments(self):
         assert_raises(TypeError, f, d, axis="invalid")
         assert_raises(TypeError, f, d, axis="invalid", dtype=None,
                       keepdims=True)
+        assert_raises_regex(TypeError, "integer or tuple", f, d, axis=[0])
```

Comment:
it comes from the PyNumber_Index here https://github.com/numpy/numpy/blob/master/numpy/core/src/multiarray/conversion_utils.c#L848